src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -372,4 +390,119 @@ private OffenderCell transform(final AgencyInternalLocation cell, final boolean
             .attributes(attributes)
             .build();
     }
+
+    @Transactional
+    public AddressDto createAgencyAddress(final String agencyId, final RequestToUpdateAddress requestToUpdateAddress) {
+        final var agency = agencyLocationRepository.findById(agencyId).orElseThrow(EntityNotFoundException.withId(agencyId));
+
+        final AgencyAddress address = AgencyAddress.builder()
+            .flat(requestToUpdateAddress.getFlat())
+            .premise(requestToUpdateAddress.getPremise())
+            .street(requestToUpdateAddress.getStreet())
+            .locality(requestToUpdateAddress.getLocality())
+            .postalCode(requestToUpdateAddress.getPostalCode())
+            .primaryFlag(requestToUpdateAddress.isPrimary() ? "Y" : "N")
+            .noFixedAddressFlag(requestToUpdateAddress.isNoFixedAddress() ? "Y" : "N")
+            .startDate(requestToUpdateAddress.getStartDate())
+            .endDate(requestToUpdateAddress.getEndDate())
+            .commentText(requestToUpdateAddress.getComment())
+            .addressType(addressTypeReferenceCodeRepository.findById(AddressType.pk(requestToUpdateAddress.getAddressType())).orElseThrow(EntityNotFoundException.withId(requestToUpdateAddress.getAddressType())))
+            .city(StringUtils.isNotBlank(requestToUpdateAddress.getTown()) ? cityReferenceCodeRepository.findById(City.pk(requestToUpdateAddress.getTown())).orElseThrow(EntityNotFoundException.withId(requestToUpdateAddress.getTown())) : null)
```

Comment:
So the logic of mapping the COUNTY etc to the correct code will be in the register-to-nomis service, which makes sense

Reply:
yes